src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -185,6 +193,44 @@ private StatementSyntax Transform( StatementSyntax stmt )
 			_ => UnhandledSyntax( stmt )
 		};
 
+	private StatementSyntax Transform( ReturnStatementSyntax returnStmt ) {
+		var expr = returnStmt.Expression;
+		if( !m_generatedFunctionReturnsVoid || expr is null ) {
+			return returnStmt.WithExpression( MaybeTransform( returnStmt.Expression, Transform ) );
+		}
+
+		if( IsStatementCompatibleExpression( expr ) == Compatibility.Compatible ) {
+			return SyntaxFactory.Block(
+					SyntaxFactory.ExpressionStatement( Transform( expr ) ).WithLeadingTrivia( SyntaxFactory.Space ),
+					SyntaxFactory.ReturnStatement().WithLeadingTrivia( SyntaxFactory.Space ).WithTrailingTrivia( SyntaxFactory.Space )
+			).WithTriviaFrom( returnStmt );
+		} else if (IsStatementCompatibleExpression( expr ) == Compatibility.Incompatible ) {
+			return SyntaxFactory.ReturnStatement().WithTriviaFrom( returnStmt );
+		}
+
+		return UnhandledSyntax( returnStmt );

```

Comment:
```suggestion
		return IsStatementCompatibleExpression( expr ) switch {
			Compatibility.Compatible => SyntaxFactory.Block(
					SyntaxFactory.ExpressionStatement( Transform( expr ) ).WithLeadingTrivia( SyntaxFactory.Space ),
					SyntaxFactory.ReturnStatement().WithLeadingTrivia( SyntaxFactory.Space ).WithTrailingTrivia( SyntaxFactory.Space )
			).WithTriviaFrom( returnStmt ),
			Compatibility.Incompatible => SyntaxFactory.ReturnStatement().WithTriviaFrom( returnStmt ),
			Compatibility.Unsupported => UnhandledSyntax( returnStmt ),
			_ => throw new NotImplementedException()
		};
```

(switch makes it easier to see how we handle each case. The `_` case should never get hit in practice but enums have some weird sharp corners so crashing here is reasonable.)